tools/actions_local_runner.py

```diff
@@ -289,6 +290,35 @@ async def full(self) -> None:
         )
 
 
+class ClangTidy(Check):
+    name = "clang-tidy: Run clang-tidy"
+    common_options = [
+        "--clang-tidy-exe",
+        ".clang-tidy-bin/clang-tidy",
+        "--parallel",
+        "--verbose",
+    ]
+
+    def filter_files(self, files: List[str]) -> List[str]:
+        return self.filter_ext(files, {".c", ".cc", ".cpp"})
+
+    async def quick(self, files: List[str]) -> CommandResult:
+        return await shell_cmd(
+            [sys.executable, "tools/linter/clang_tidy", "--paths"]
+            + [os.path.join(REPO_ROOT, f) for f in files]
+            + self.common_options,
+        )
+
+    async def full(self) -> None:
+        await shell_cmd(
+            [
+                sys.executable,
+                "tools/linter/clang_tidy",
```

Comment:
It looks like this script always returns `0` as its exit code, so even if there are failures the runner still reports that it passed (and it doesn't print the output when the command passes). Can you fix up the clang-tidy script to thread through the return code + unify them from parallel runs?

Reply:
`clang-tidy` returns a zero exit code even if there are warnings. To resolve this comment, I'm using the `extract_warnings` method to detect if there are any warnings, and then return the number of warnings as the exit code.